posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc

```diff
@@ -0,0 +1,711 @@
+      "formula": "$5500053415/1$",
+      "power": 7,
+      "exportEnergy": 900000,
+      "importEnergy": 277783200,
+      "phaseId": 0,
+      "current": 1,
+      "apparentPower": 21,
+      "cosPhi": 33
+    },    
+    <elided>
+  ],
+  "voltages": [
+    {
+      "voltage": 242,
+      "phaseId": 0
+    },
+    <elided>
+  ]
+}
+```
+
+The _Smappee Genius_ collects more information than its smaller sibling. Observe the `measuredFrequency` measurement (expressed in µHz) which allows us to track the mains AC frequency, `channelPowers.cosPhi`, measures the so-called <<cos-phi>> or power factor on a per channel basis. Interesting to note is the presence of the _CT Hub_ Id and channel number in the `channelPowers.formula` value. This Id is a 10-digit number that uniquely identifies each _CT Hub_.
+
+[id=smappee-infinity-connection-diagram]
+The following diagram shows the physical configuration & measurement points on the
+_Smappee Infinity_ system.
+
+image::smappee-connection-diagram.png[Smappee connection diagram]
+
+We recognise the four _CT Hubs_ with their respective Id and what each _CT Hub_ channel measures. The labels _Ground Floor_ and _2nd Floor_ at the top of the diagram refer to the location of the fuse panel in which the _Smappee_ components are located.
+
+Now that we know the message format of the raw data published via _MQTT_, we will look at how we can get the messages into the Time series database.
+
+=== Configuring Telegraf
+
+Telegraf offers a series of plugins that fall into four categories: _Input_, _Processor_, _Aggregator_, and _Output_. Telegraf plugins that are relevant to our use case are:
+
+* Input: https://docs.influxdata.com/telegraf/v1.26/plugins/#input-mqtt_consumer[MQTT Consumer] and the https://docs.influxdata.com/telegraf/v1.21/data_formats/input/json_v2[JSON v2 parser]. The JSON v2 parser is a generic component that can apply to any input plugin.
+
+* Aggregator: https://docs.influxdata.com/telegraf/v1.26/plugins/#aggregator-basicstats[Basic Stats]
+
+* Processor: https://docs.influxdata.com/telegraf/v1.26/plugins/#processor-regex[Regex] 
+
+* Output: https://docs.influxdata.com/telegraf/v1.23/plugins/#output-influxdb[InfluxDB v1.x]
+
+==== Reading and transforming the MQTT data sources
+
+Our two _Smappees_ send data to the _EMQX_ _MQTT_ broker located at `tcp://192.168.68.201:1883`. Let’s look at the relevant part of the Telegraf configuration for the _Smappee Genius_.
+
+```toml
+[[inputs.mqtt_consumer]]
+  alias = "smappee-2"
+  name_override = "smappee-data-2"
+  servers = ["tcp://192.168.68.201:1883"]
+  topics = [
+    "servicelocation/5aaf6e89-89cb-4e33-bf34-05abc62f5563/realtime"
+  ]
+  # The "host" tag is irrelevant in this use case. Drop it
+  tagexclude = ["host"]
+  data_format = "json_v2"
+  [[inputs.mqtt_consumer.json_v2]]
+    [[inputs.mqtt_consumer.json_v2.field]]
+      path = "channelPowers.#(formula==$5500048161/0$).power"
+      rename = "zolder-verlichting"
```

Comment:
can you use the same english names you used above? `lighting` , `bicycle garage`, `sockets-network-switch`?

Reply:
Will do.